source-chargebee-native/source_chargebee_native/api.py

```diff
@@ -208,12 +220,15 @@ async def fetch_resource_changes(
     assert isinstance(log_cursor, int)
 
     start_date = _ts_to_dt(log_cursor)
-    end_date = min(start_date + timedelta(days=30), datetime.now(tz=UTC))
+    # Apply LAG to handle eventual consistency - don't fetch records more recent than LAG ago
+    now_with_lag = datetime.now(tz=UTC) - LAG
+    end_date = min(start_date + timedelta(days=30), now_with_lag)
```

Comment:
FYI @Alex-Bair  - I will be force pushing to change the places where we use an arbitrary 30 day window for when the `log_cursor` is in the past due to a capture being disabled and re-enabled at a later date or some other reason. This has shown to take quite some time for large streams like `Transactions` and `Invoices`.

I will set this to: `end_date = min(start_date + MIN_INCREMENTAL_INTERVAL, now_with_lag)` so that we make incremental progress instead. This could have some impact on performance (querying smaller ranges), so I will try to monitor that.